src/main/java/reposense/parser/SinceDateArgumentType.java

```diff
@@ -25,8 +27,12 @@
      */
     @Override
     public Optional<Date> convert(ArgumentParser parser, Argument arg, String value) throws ArgumentParserException {
-        return (FIRST_COMMIT_DATE_SHORTHAND.equals(value))
-                ? Optional.of(ARBITRARY_FIRST_COMMIT_DATE)
-                : super.convert(parser, arg, value);
+        if (FIRST_COMMIT_DATE_SHORTHAND.equals(value)) {
+            return Optional.of(ARBITRARY_FIRST_COMMIT_DATE);
+        } else {
+            Matcher matcher = Pattern.compile(DATE_FORMAT_REGEX).matcher(value);
+            String sinceDate = matcher.find() ? matcher.group(1) : value;
```

Comment:
Can move these two lines to parent class to avoid duplication.

Also `Match#group` may throw `IndexOutOfBoundsException`.

Finally, can remove the else block. See [happy path](https://nus-cs2103-ay1819s1.github.io/cs2103-website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent).

Reply:
Might not be able to move the two lines to parent class as the matching of the first substring that matches the date format must happen before appending the time (`00:00:00` for since dates and `23:59:59` for until dates)

Reply:
Don't have be in `DateArgumentType#convert`, just make a new protected method?

Reply:
Oh ya, okay got it thanks so much.

Reply:
@yong24s 
Do we need to try catch the `IndexOutOfBoundsException` from `matcher.group`? Since we already have a `matcher.find()` before `matcher.group(1)`, which ensures that the matcher will have at least a match.